block/sync_service.go

```diff
@@ -26,27 +26,40 @@ import (
 	"github.com/rollkit/rollkit/types"
 )
 
-// BlockSyncService is the P2P Sync Service for block that implements the
-// go-header interface.  Contains a block store where synced blocks are stored.
+// SyncService is the P2P Sync Service for blocks and headers.
+//
 // Uses the go-header library for handling all P2P logic.
-type BlockSyncService struct {
-	conf       config.NodeConfig
-	genesis    *cmtypes.GenesisDoc
-	p2p        *p2p.Client
-	ex         *goheaderp2p.Exchange[*types.Block]
-	sub        *goheaderp2p.Subscriber[*types.Block]
-	p2pServer  *goheaderp2p.ExchangeServer[*types.Block]
-	blockStore *goheaderstore.Store[*types.Block]
-
-	syncer       *goheadersync.Syncer[*types.Block]
+type SyncService[H header.Header[H]] struct {
+	conf      config.NodeConfig
+	genesis   *cmtypes.GenesisDoc
+	p2p       *p2p.Client
+	ex        *goheaderp2p.Exchange[H]
+	sub       *goheaderp2p.Subscriber[H]
+	p2pServer *goheaderp2p.ExchangeServer[H]
+	store     *goheaderstore.Store[H]
+	prefix    string
+
+	syncer       *goheadersync.Syncer[H]
 	syncerStatus *SyncerStatus
 
 	logger log.Logger
 	ctx    context.Context
 }
 
+type BlockSyncService = SyncService[*types.Block]
+type HeaderSyncService = SyncService[*types.SignedHeader]
+
 // NewBlockSyncService returns a new BlockSyncService.
 func NewBlockSyncService(ctx context.Context, store ds.TxnDatastore, conf config.NodeConfig, genesis *cmtypes.GenesisDoc, p2p *p2p.Client, logger log.Logger) (*BlockSyncService, error) {
+	return newSyncService[*types.Block](ctx, store, "block", conf, genesis, p2p, logger)
+}
+
+// NewHeaderSyncService returns a new HeaderSyncService.
+func NewHeaderSyncService(ctx context.Context, store ds.TxnDatastore, conf config.NodeConfig, genesis *cmtypes.GenesisDoc, p2p *p2p.Client, logger log.Logger) (*HeaderSyncService, error) {
+	return newSyncService[*types.SignedHeader](ctx, store, "header", conf, genesis, p2p, logger)
```

Comment:
with this update it might be worth creating a `type syncType string` for some type safety and removal of bare `string + string` operations. 

imo it also makes a little more sense than calling it a prefix.